sycl/include/sycl/detail/type_traits.hpp

```diff
@@ -50,6 +50,11 @@ template <class T>
 inline constexpr bool is_user_constructed_group_v =
     is_user_constructed_group<T>::value;
 
+template <class T> struct is_fixed_size_group : std::false_type {};
+
+template <class T>
+inline constexpr bool is_fixed_size_group_v = is_fixed_size_group<T>::value;
```

Comment:
I'd prefer it in `detail::`.  I missed this during my review -- I thought that because this header is in `detail`, everything was in the `detail` namespace.  If that's not true, it would be nice if somebody (outside of this PR) could split them out.